src/manager/YaraManager.js

```diff
@@ -0,0 +1,124 @@
+import yara from 'yara';
+import { promisify } from 'util';
+
+export default class YaraManager {
+  constructor() {
+    this.resources = [];
+  }
+
+  /*
+   * This Method scans a list of raw js artifacts using
+   * a set of Yara rules. Returns the Js artifacts that match
+   * otherwise, no malware was detected.
+   *
+   * If an error occurs in processing, values are pushed to the RO
+   * in order to be handled correctly by our api.
+   *
+   * When pushing errors to the RO data buffer:
+   *      error: boolean (whether or not an error has occurred in processing)
+   *      matchedRule: only valid if (isMalicious == true)
+   *      isMalicious: boolean, null if unknown
+   */
+  async setupResourceScan(jsArtifacts, ticket) {
+    console.log('Scanning JS...');
+    const ticketId = ticket.getID();
+    const responseFile = 'yara_response.txt';
+    const initialize = promisify(yara.initialize);
+
+    try {
+      // link to yara binary
+      await initialize();
+
+      const rules = [{ filename: 'src/resources/rules.yara' }];
+      const scanner = yara.createScanner();
+      const warnings = await new Promise((resolve, reject) =>
```

Comment:
I am kind of curious... Why didn't you promisify scanner.configure here but rather create a Promise from scratch?

Reply:
Hoo boy - you found the ugliest code in the PR. It's got to do with `this` context and `node-gyp`. Because this module is using some natives to the native Yara C library, it generates some sort of special, not-javascript-but-still-readable-by-node.js-code. If you go to the `yara` module in your IDE, and start asking it for definitions, it'll give up pretty quickly and point you to the `yara.node` file, which gets dynamically generated when the module is compiled on your computer.

Node internally is using some sort of weird pointer magic to make this happen, pointing `this.yara` to an instance of the real module bindings. When using `util.promisify`, this link is broken - and anything under `this.yara` fails to resolve. I'm not sure what `util.promisify` is doing, but because it's trying to reshape/rewrite some functions at runtime, I could see it getting confused by the other injection that node.js is doing. You'd think they'd handle that, since both features are part of the language spec.

I initially wrapped only one in a `Promise`, because it was on a member function of a class. `util.promisify` doesn't work on instances (of classes), or entire APIs. Some more invasive/complex dependencies like [bluebird](http://bluebirdjs.com/docs/features.html) allow you to wrap entire modules in `Promise`s instead of callbacks, but they also implement their own promise APIs in userspace, and are huge and hard-to-understand dependencies in their own right. So I didn't experiment with pulling one of those in, although it might be able to handle it. Certainly would look cleaner, at the expense of a lot of behind-the-scenes complexity.

Hope that helps. It's a really messy thing that's going here, and I tried to hide it in a ternary expression in the hopes that readers would find it easy to gloss over :P